### DISCUSSION TITLE
Tags not created

### DISCUSSION THREAD

**Discussion by JesusMF23** [PaulHatch/semantic-version#134] [Q&A]
Hello!

I am using the github action but I have an issue. Although on logs of my workflow I get printed the version and the link to create a release, it actually didn't create any tag on my repository.

Meaning, I always need to go to the logs and click the generated link to create a release and therefore the tag on my repo.

How can I get my tag created without the need to create a release?

here is my setup:

```
 steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      with:
        fetch-depth: 0
      
    - name: Create version tags
      uses: paulhatch/semantic-version@v5.3.0
      with:
        # The prefix to use to identify tags
        tag_prefix: "v"
        # A string which, if present in a git commit, indicates that a change represents a
        # major (breaking) change, supports regular expressions wrapped with '/'
        major_pattern: "(MAJOR)"
        # A string which indicates the flags used by the `major_pattern` regular expression. Supported flags: idgs
        major_regexp_flags: ""
        # Same as above except indicating a minor change, supports regular expressions wrapped with '/'
        minor_pattern: "(MINOR)"
        # A string which indicates the flags used by the `minor_pattern` regular expression. Supported flags: idgs
        minor_regexp_flags: ""
        # A string to determine the format of the version output
        version_format: "${major}.${minor}.${patch}-prerelease${increment}"
        # Optional path to check for changes. If any changes are detected in the path the
        # 'changed' output will true. Enter multiple paths separated by spaces.
        change_path: "functions"
        # Named version, will be used as suffix for name version tag
        namespace: interactions
        # If this is set to true, *every* commit will be treated as a new version.
        bump_each_commit: true
        # If bump_each_commit is also set to true, setting this value will cause the version to increment only if the pattern specified is matched.
        bump_each_commit_patch_pattern: "(PATCH)"
        # If true, the body of commits will also be searched for major/minor patterns to determine the version type.
        search_commit_body: true
        # The output method used to generate list of users, 'csv' or 'json'.
        user_format_type: "json"
        # Prevents pre-v1.0.0 version from automatically incrementing the major version.
        # If enabled, when the major version is 0, major releases will be treated as minor and minor as patch. Note that the version_type output is unchanged.
        enable_prerelease_mode: false
        # If enabled, diagnostic information will be added to the action output.
        debug: true
        # If true, the branch will be used to select the maximum version.
        version_from_branch: false
```

**Comment by PaulHatch** [comment]
The primary mode, version from tags, is intended primarily for scenarios where software is tested before being released and assumes that tagging is part of a formal release strategy with possible additional gates (QA/UAT, etc) which are not automated.

If you want to increment the version on every build, consider using the `bump_each_commit` option which does not require tags to increment the version. There is a [guide](https://github.com/PaulHatch/semantic-version/blob/master/guide.md) that covers various approaches to versioning that may be useful as well. In any case tagging automatically is not recommended, if you decide to do this you should set the max concurrency to 1 to prevent race conditions.

**Reply by JesusMF23** [reply]
Hi Paul!

Thanks for your answer! I am using the bump_each_commit, is set to true and I look for (PATCH). The thing that I mention is that, although I see on the logs the version that is supposed to have my service, there is no real impact on my code.

I mean, there is no automated tag created on my repository or any package.json upgrading the version or anything.

For a propper management of versions we need to have automated tags for each version as part of the CICD, if something fails on 1.0.15 but was working on 1.0.14, QA team should be able to point out the version afterwards and point to specific tag in github.

what I mean is, under tags on my repo I only see the one created manually by me before using this action:

![image](https://github.com/PaulHatch/semantic-version/assets/58133399/1540e093-c982-45d6-bd60-b6c6f64762d9)

is there any way to create the tags (not the releases) automatically?

**Reply by JesusMF23** [reply]
I have another question, taking into account the previous image where I already have a tag "v0.7.0-beta", why this code is not recognizing that previous tag?

```
    - name: Checkout repository
      uses: actions/checkout@v2
      with:
        fetch-depth: 0
      
    - name: Create version tags
      uses: paulhatch/semantic-version@v5.3.0
      with:
        # The prefix to use to identify tags
        tag_prefix: "v"
        # A string which, if present in a git commit, indicates that a change represents a
        # major (breaking) change, supports regular expressions wrapped with '/'
        major_pattern: "(MAJOR)"
        # A string which indicates the flags used by the `major_pattern` regular expression. Supported flags: idgs
        major_regexp_flags: ""
        # Same as above except indicating a minor change, supports regular expressions wrapped with '/'
        minor_pattern: "(MINOR)"
        # A string which indicates the flags used by the `minor_pattern` regular expression. Supported flags: idgs
        minor_regexp_flags: ""
        # A string to determine the format of the version output
        version_format: "${major}.${minor}.${patch}-beta"
        # Optional path to check for changes. If any changes are detected in the path the
        # 'changed' output will true. Enter multiple paths separated by spaces.
        change_path: "functions"
        # Named version, will be used as suffix for name version tag
        namespace: "interactions"
        # If this is set to true, *every* commit will be treated as a new version.
        bump_each_commit: true
        # If bump_each_commit is also set to true, setting this value will cause the version to increment only if the pattern specified is matched.
        bump_each_commit_patch_pattern: ""
        # If true, the body of commits will also be searched for major/minor patterns to determine the version type.
        search_commit_body: false
        # The output method used to generate list of users, 'csv' or 'json'.
        user_format_type: "json"
        # Prevents pre-v1.0.0 version from automatically incrementing the major version.
        # If enabled, when the major version is 0, major releases will be treated as minor and minor as patch. Note that the version_type output is unchanged.
        enable_prerelease_mode: false
        # If enabled, diagnostic information will be added to the action output.
        debug: false
        # If true, the branch will be used to select the maximum version.
        version_from_branch: true
```

I am getting this warning and version:
![image](https://github.com/PaulHatch/semantic-version/assets/58133399/9cbf3fc6-79c3-464b-938e-854d0ade954e)

**Comment by PaulHatch** [comment]
This action provides information only. There is no impact on the code unless you use the outputs. You can use this in your build to generate the output artifacts, for example to tag a Docker image. Certain languages have an ambient project version as well which can pretty much always be set from the command line, you can use the output of the action here as well. In any case the specifics depend on your project type.

Regarding tags, the configuration you have above is looking for tags like `v1.2.3-interactions` since you've added a namespace. If you want to tag each commit as well you would do this in a subsequent action, there are a number of GitHub actions available to tag out there. If the tag does not match the pattern expected it will be ignored, but this may be fine for your case if you are only looking to identify the commit for a particular version. Since you are just trying to identify the commit, consider also using the git hash as part of your version in this case, e.g. `1.2.3-30186c8`, it may be easier.

I see you are using version from branch as well, this has additional implications if building on a version branch described in the guide.

**Reply by JesusMF23** [reply]
Got it thanks!

Now I have this other issue, I still get an incorrect version, I have a tag with v0.7.0-beta (manually created, the one on the screenshot above) but I get version 0.0.79 and then the release link with the corresponding v0.0.79-beta

```
    - name: Checkout repository
      uses: actions/checkout@v2
      with:
        fetch-depth: 0
      
    - name: Create version tags
      uses: paulhatch/semantic-version@v5.3.0
      with:
        # The prefix to use to identify tags
        tag_prefix: "v"
        # A string which, if present in a git commit, indicates that a change represents a
        # major (breaking) change, supports regular expressions wrapped with '/'
        major_pattern: "(MAJOR)"
        # A string which indicates the flags used by the `major_pattern` regular expression. Supported flags: idgs
        major_regexp_flags: ""
        # Same as above except indicating a minor change, supports regular expressions wrapped with '/'
        minor_pattern: "(MINOR)"
        # A string which indicates the flags used by the `minor_pattern` regular expression. Supported flags: idgs
        minor_regexp_flags: ""
        # A string to determine the format of the version output
        version_format: "${major}.${minor}.${patch}"
        # Optional path to check for changes. If any changes are detected in the path the
        # 'changed' output will true. Enter multiple paths separated by spaces.
        change_path: "functions"
        # Named version, will be used as suffix for name version tag
        namespace: "beta"
        # If this is set to true, *every* commit will be treated as a new version.
        bump_each_commit: true
        # If bump_each_commit is also set to true, setting this value will cause the version to increment only if the pattern specified is matched.
        bump_each_commit_patch_pattern: ""
        # If true, the body of commits will also be searched for major/minor patterns to determine the version type.
        search_commit_body: false
        # The output method used to generate list of users, 'csv' or 'json'.
        user_format_type: "json"
        # Prevents pre-v1.0.0 version from automatically incrementing the major version.
        # If enabled, when the major version is 0, major releases will be treated as minor and minor as patch. Note that the version_type output is unchanged.
        enable_prerelease_mode: false
        # If enabled, diagnostic information will be added to the action output.
        debug: false
        # If true, the branch will be used to select the maximum version.
        version_from_branch: false
```

here you can check the outputs, can you help me? what is wrong in my config? it should be something like version: v0.7.79-beta, am I correct?

<img width="1083" alt="image" src="https://github.com/PaulHatch/semantic-version/assets/58133399/5e1ed254-b383-472f-bc83-dd69d3935e77">

**Reply by PaulHatch** [reply]
Can't tell from here. You have specified a change path, functions, does the tag appear on a commit which changes files on this change path?

There is a debug mode which allows you to output diagnostics information, specifically each git command executed and the result received. You could check there what is being returned for the tags call.
